Added CMORizer for Landschützer2020 (spco2)

schlunma opened the pull request:
<!--
    Thank you for contributing to our project!

    Please do not delete this text completely, but read the text below and keep
    items that seem relevant. If in doubt, just keep everything and add your
    own text at the top, a reviewer will update the checklist for you.

    While the checklist is intended to be filled in by the technical and scientific
    reviewers, it is the responsibility of the author of the pull request to make
    sure all items on it are properly implemented.
-->

## Description

This PR adds a CMORizer for the variable `spco2` (Omon) of the [Landschützer2020](https://www.ncei.noaa.gov/access/metadata/landing-page/bin/iso?id=gov.noaa.nodc%3A0209633) dataset.

In addition, a couple of Codacy issues in `utilities.py` have been fixed.

<!--
    Please describe your changes here, especially focusing on why this pull request makes
    ESMValTool better and what problem it solves.

    Before you start, please read our contribution guidelines: https://docs.esmvaltool.org/en/latest/community/

    Please fill in the GitHub issue that is closed by this pull request, e.g. Closes #1903
-->

- Link to documentation: https://esmvaltool--2908.org.readthedocs.build/en/2908/input.html#supported-datasets-for-which-a-cmorizer-script-is-available

* * *

## Checklist

It is the responsibility of the author to make sure the pull request is ready to review. The icons indicate whether the item will be subject to the [🛠 Technical][1] or [🧪 Scientific][2] review.

<!-- The next two lines turn the 🛠 and 🧪 below into hyperlinks -->
[1]: https://docs.esmvaltool.org/en/latest/community/review.html#technical-review
[2]: https://docs.esmvaltool.org/en/latest/community/review.html#scientific-review

- [x] [🛠][1] This pull request has a [descriptive title](https://docs.esmvaltool.org/en/latest/community/code_documentation.html#pull-request-title)
- [x] [🛠][1] Code is written according to the [code quality guidelines](https://docs.esmvaltool.org/en/latest/community/code_documentation.html#code-quality)
- [x] [🛠][1] [Documentation](https://docs.esmvaltool.org/en/latest/community/code_documentation.html#documentation) is available
- [x] [🛠][1] [Tests](https://docs.esmvaltool.org/en/latest/community/code_documentation.html#tests) run successfully
- [x] [🛠][1] The [list of authors](https://docs.esmvaltool.org/en/latest/community/code_documentation.html#list-of-authors) is up to date
- [x] [🛠][1] Any changed dependencies have been [added or removed correctly](https://docs.esmvaltool.org/en/latest/community/code_documentation.html#dependencies)
- [x] [🛠][1] All [checks below this pull request](https://docs.esmvaltool.org/en/latest/community/code_documentation.html#pull-request-checks) were successful

### [New or updated data reformatting script](https://docs.esmvaltool.org/en/latest/develop/dataset.html)

- [x] [🛠][1] [Documentation](https://docs.esmvaltool.org/en/latest/community/dataset.html#dataset-documentation) is available
- [x] [🛠][1] The dataset has been [added to the CMOR check recipe](https://docs.esmvaltool.org/en/latest/community/dataset.html#testing)
- [x] [🧪][2] Numbers and units of the data look [physically meaningful](https://docs.esmvaltool.org/en/latest/community/dataset.html#scientific-sanity-check)

***

To help with the number of pull requests:

-  🙏 We kindly ask you to [review](https://docs.esmvaltool.org/en/latest/community/review.html#review-of-pull-requests) two other [open pull requests](https://github.com/ESMValGroup/ESMValTool/pulls) in this repository

<!--
If you need help with any of the items on the checklists above, please do not hesitate to ask by commenting in the issue or pull request.
-->

valeriupredoi reviewed (approved):
nice one - technically reviewed and approved too, and thanks for fixing a bunch of ye olde crappe code in utils too :beer:

remi-kazeroni commented on esmvaltool/cmorizers/data/utilities.py:
```diff
@@ -58,6 +58,22 @@ def add_height10m(cube):
 
 
+def add_scalar_depth_coord(cube, depth=0.0):
+    """Add scalar coordinate 'depth' with value of `depth`m."""
```
You might want to expand this docstring as done above.

remi-kazeroni reviewed (commented):
Thanks for your contribution and cleaning `utilities.py` @schlunma! This PR looks good to me. Everything works fine: downloader, formatter, recipe_check_obs 👍  I've put the raw and CMORized data in our shared Levante directories.

It seems we already have support for an older version of the dataset (Landschuetzer2016) for the variable `spco2`. Could you please briefly comment in this PR if that support should be kept or not and if the recipes using the "2016" version of the data should be updated? Thanks!

schlunma commented:
Thanks for the review @remi-kazeroni!

I cleaned all docstrings for these functions and added one for the new function.

Regarding the Landschuetzer2016 datasets: This is not really an older version of the dataset. The Landschuetzer2016 dataset contains monthly mean data from 1982 to 2015, while this new Landschuetzer2020 dataset contains a monthly climatology for the years 1988-2020 (i.e., 12 values where the first corresponds to the January average over all years, the second to the February average over all years, etc.). So no, the dataset should not be replace in the recipe :smile:

remi-kazeroni reviewed (approved):
Thanks for the changes and the clarification @schlunma! Indeed, no confusion possible between the 2 datasets 👍 I will merge this at the end of the week if no further comments.

remi-kazeroni commented:
@schlunma: could you please merge the `main` branch into this one so that I could merge the PR? This is needed since I just merged another CMORizer PR 😎

remi-kazeroni commented:
Hmm, the docs build is now failing due to some gdal issues... I'm not really sure why because it works fine for me locally using an environment with the same `gdal` and `libgdal` versions. Any idea what happened @schlunma?  Is it worth trying another "merge main branch" since the docs build cannot be retriggered directly?

schlunma commented:
No idea what's happening there...I'll try merging main into this. I already tried to rebuild the doc again, and it failed for the same reason. :shrug:

schlunma commented:
Now the tests pass :tada:
